internals/overlord/logstate/loki/loki.go

```diff
@@ -52,11 +58,19 @@ func NewClient(target *plan.LogTarget) *Client {
 }
 
 func (c *Client) Add(entry servicelog.Entry) error {
+	if N := len(c.entries); N >= maxRequestEntries {
+		// make room for 1 entry
+		c.entries = c.entries[(N - maxRequestEntries + 1):]
+	}
+	if cap(c.entries)-len(c.entries) == 0 {
```

Comment:
Sorry just looking at some tests. We don't need this if statement and the realloc. I think the original code is fine due to the behaviour of append.

Reply:
I was mistakingly looking at the wrong memory statistic in my own testing.

Reply:
yep, append does the reallocation for us as needed